fe/fe-core/src/main/java/com/starrocks/qe/SetDefaultRoleExecutor.java

```diff
@@ -40,10 +39,9 @@ public static void execute(SetDefaultRoleStmt stmt, ConnectContext context) thro
             roleIds = new HashSet<>();
             for (String roleName : stmt.getRoles()) {
                 Long roleId = manager.getRoleIdByNameAllowNull(roleName);
-                if (roleId == null || !roleIdsForUser.contains(roleId)) {
-                    throw new SemanticException("Role " + roleName + " is not granted to " + user.toString());
+                if (roleId != null && roleIdsForUser.contains(roleId)) {
```

Comment:
So, we won't prompt an error if user set an unowned role as his default role, why we change this logic?

Reply:
got it, it's moved to the analyisis phase